File: app/src/main/java/com/moggot/findmycarlocation/about/AboutFragment.java
package com.moggot.findmycarlocation.about;

import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;
import android.view.View;
import android.widget.TextView;

import com.moggot.findmycarlocation.BuildConfig;
import com.moggot.findmycarlocation.MainActivity;
import com.moggot.findmycarlocation.R;
import com.moggot.findmycarlocation.base.BaseFragment;
import com.moggot.findmycarlocation.di.component.MainComponent;

import org.jetbrains.annotations.NotNull;

import java.util.Calendar;

public class AboutFragment extends BaseFragment {

    private static final String TAG = "AboutFragment";

    public static AboutFragment newInstance() {
        return new AboutFragment();
    }

    @Override
    public String getFragmentTag() {
        return TAG;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        TextView tvVersion = view.findViewById(R.id.about_tv_version);
        TextView tvCopyright = view.findViewById(R.id.about_tv_copyright);
        ConstraintLayout clPrivacyPolicy = view.findViewById(R.id.about_cl_privacy_policy);
        ConstraintLayout clRemoveAds = view.findViewById(R.id.about_cl_purchase_premium);

        tvVersion.setText(getString(R.string.version, BuildConfig.VERSION_NAME));
        tvCopyright.setText(getString(R.string.copyright, Calendar.getInstance().get(Calendar.YEAR)));
        MainActivity activity = ((MainActivity) getActivity());
        if (activity == null) {
            return;
        }
        if (activity.isPremiumPurchased()) {
            clRemoveAds.setVisibility(View.GONE);
        }
        clPrivacyPolicy.setOnClickListener(v -> {
            startActivity(new Intent(getContext(), PrivacyPolicyActivity.class));
        });
        clRemoveAds.setOnClickListener(v -> activity.getBillingManager().requestSubscription());
    }

    @Override
    public int getLayoutRes() {
        return R.layout.fragment_about;
    }

    @NotNull
    @Override
    public String getComponentName() {
        return MainComponent.class.getName();
    }

    @Override
    public boolean isComponentDestroyable() {
        return false;
    }
}

File: app/src/main/java/com/moggot/findmycarlocation/about/AboutFragment.kt
package com.moggot.findmycarlocation.about

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.View.GONE
import android.view.View.VISIBLE
import androidx.fragment.app.viewModels
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import com.moggot.findmycarlocation.BuildConfig
import com.moggot.findmycarlocation.MainActivity
import com.moggot.findmycarlocation.R
import com.moggot.findmycarlocation.base.viewBinding
import com.moggot.findmycarlocation.common.BaseFragment
import com.moggot.findmycarlocation.databinding.FragmentAboutBinding
import kotlinx.coroutines.launch
import java.util.Calendar

class AboutFragment : BaseFragment(R.layout.fragment_about) {

    override val viewModel by viewModels<AboutViewModel>()
    override val fragmentTag: String = "AboutFragment"
    private val viewBinding by viewBinding(FragmentAboutBinding::bind)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewBinding.aboutTvVersion.text = getString(R.string.version, BuildConfig.VERSION_NAME)
        viewBinding.aboutTvCopyright.text = getString(
            R.string.copyright,
            Calendar.getInstance()[Calendar.YEAR]
        )
        val mainActivity: MainActivity = activity as? MainActivity ?: return
        if ((activity as MainActivity).isPremiumPurchased) {
            viewBinding.aboutClPurchasePremium.visibility = View.GONE
        }
        viewBinding.aboutClPrivacyPolicy.setOnClickListener {
            startActivity(Intent(context, PrivacyPolicyActivity::class.java))
        }
        viewLifecycleOwner.lifecycleScope.launch {
            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
                mainActivity.billingManager.isPremium.collect { isPremium ->
                    viewBinding.aboutClPurchasePremium.visibility = if (isPremium) GONE else VISIBLE
                }
            }
        }
        viewBinding.aboutClPurchasePremium.setOnClickListener { v -> mainActivity.billingManager.requestSubscription() }
    }

    companion object {
        fun newInstance(): AboutFragment {
            return AboutFragment()
        }
    }
}
